networkx/algorithms/reciprocity.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+#
+#    Copyright (C) 2015 by 
+#    Haochen Wu <wuhaochen42@gmail.com>
+#    All rights reserved.
+#    BSD license.
+"""Algorithms to calculate reciprocity in a directed graph."""
+from networkx import NetworkXError
+from ..utils import not_implemented_for
+
+__author__ = """\n""".join(['Haochen Wu <wuhaochen42@gmail.com>'])
+
+__all__= ['reciprocity','overall_reciprocity']
+
+@not_implemented_for('undirected','multigraph')
+def reciprocity(G, nodes=None):
+    """Computer the reciprocity in a directed graph.
```

Comment:
There is an example in stochastic.py where we import ```warnings``` and use ```warnings.warn```.
Most of the time we find a good value to return that indicates the warning without printing the sometimes annoying warning line at runtime. That is the idea of placing the value None instead of a number in the dict. Either way it should be documented that this is what will happen for isolated nodes (probably in a ```Notes``` section of the doc_string).